logic/src/commonMain/kotlin/com/wire/kalium/logic/data/team/TeamRepository.kt

```diff
@@ -0,0 +1,41 @@
+package com.wire.kalium.logic.data.team
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.logic.wrapApiRequest
+import com.wire.kalium.network.api.teams.TeamsApi
+import com.wire.kalium.persistence.dao.TeamDAO
+import kotlinx.coroutines.flow.firstOrNull
+
+interface TeamRepository {
+    suspend fun syncSelfTeam(): Either<CoreFailure, Unit>
+}
+
+class TeamDataSource(
```

Comment:
I'm wondering if this implementation could be internal also, as I don't see cases where we would use directly this repository outside Kalium